addons/stock/migrations/9.0.1.1/post-migration.py

```diff
@@ -35,21 +35,23 @@ def _migrate_pack_operation(env):
     mark pickings that need to recreate pack operations, and update new field
     qty_done on stock.pack.operation for transferred pickings.
     """
-    env.cr.execute(
-        "select o.id, o.%(lot_id)s, p.state, o.qty_done "
-        "from stock_pack_operation o "
-        "join stock_picking p on o.picking_id=p.id "
-        "where o.%(lot_id)s is not null "
-        "group by o.id, o.%(lot_id)s, p.state, o.qty_done",
+    openupgrade.logged_query(
+        env.cr,
+        """ INSERT INTO stock_pack_operation_lot (
+            id, lot_id, operation_id, qty_todo, qty,
+            create_uid, write_uid, create_date, write_date)
+        SELECT
+            nextval('stock_pack_operation_lot_id_seq'),
+            %(lot_id)s, o.id,
+            CASE WHEN p.state != 'done' THEN o.product_qty ELSE 0 END,
+            CASE WHEN p.state = 'done' THEN o.product_qty ELSE 0 END,
+            1, 1, now() at time zone 'UTC', now() at time zone 'UTC'
```

Comment:
Better to use here the same user as the stock_pack_operation, and even the dates (although this one is more debatable)

Reply:
Done!